### PR TITLE
feat(todoist): add app, authentication, docs

### PR DESCRIPTION
**Pull request by irlevesque**
Hi, this PR adds support for triggering on Todoist tasks and for creating new tasks as an action. There were some lint errors but they look to be unrelated to my changes.

### CODE REVIEW

**Comment by barinali** [comment]
Hey @irlevesque, 

Thanks for your first contribution to Automatisch! 🎉 I have already started reviewing this pull request. However, I'm not done with it, as it takes time and effort. Nevertheless, I'll submit my review as much as I have now and continue reviewing.

**Review by barinali — CHANGES_REQUESTED** [review]
@irlevesque, thanks once again! I've reviewed everything besides the trigger you prepared. I'll submit another review for that part once I conclude it.

Let me know if you're cool with me pushing changes to your branch for any remarks I have made*, as I'd like to support you here.

\* I have tried to provide suggestions for the remarks. I hope it eases the review on your end.

File: packages/backend/src/apps/todoist/triggers/get-tasks/get-tasks.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { IGlobalVariable } from '@automatisch/types';
+
+const getActiveTasks = async ($: IGlobalVariable) => {
+
+  let requestPath = `/tasks`;
+
+  requestPath += '?project_id=' + ($.step.parameters.projectId as string).trim();
+  requestPath += '&section_id=' + ($.step.parameters.sectionId as string).trim();
+  requestPath += '&label=' + ($.step.parameters.label as string).trim();
```
Comment (barinali) [inline packages/backend/src/apps/todoist/triggers/get-tasks/get-tasks.ts:9]: @irlevesque, I could not manage to get tasks by "label". I provided `label name` as a value (I even tried it surrounded with quotes) and tested it. However, it doesn't give an active task with the label even though I see such tasks in the API response when I do not try to filter it by `label`.

I wonder if you could help me with testing this.

Comment (barinali) [reply packages/backend/src/apps/todoist/triggers/get-tasks/get-tasks.ts:9]: I see that I was using something called "shared label" and it wasn't working. Now, I tried it with a personal label and it filters by `label`! I am fine to resolve this conversation upon this.

**Comment by barinali** [comment]
@irlevesque, I am done with the review. I'd like to ask if you'd be up to reworking the authentication with [OAuth2 protocol](https://developer.todoist.com/guides/#oauth). I'd also be happy to provide this branch with that rework.

File: packages/backend/src/apps/todoist/actions/create-task/index.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    {
+      label: 'Section ID',
+      key: 'sectionId',
+      type: 'string' as const,
+      required: false,
+      description:
+        'Add task to this section ID. Example: 15551234567',
+      variables: true,
+      value: null
+    },
```
[the remark this answers is not in the document]
Comment (irlevesque) [reply packages/backend/src/apps/todoist/actions/create-task/index.ts]: Nice, I didn't realize this was possible. One thing that would be nice would be to regenerate the dynamic data based on other form fields -- for example, when a project is selected, the project ID could be used to generate a list of section IDs for that specific project.

File: packages/backend/src/apps/todoist/triggers/get-tasks/get-tasks.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  requestPath += '&filter=' + ($.step.parameters.filter as string).trim();
+
+  const response = await $.http.get(requestPath);
```
[the remark this answers is not in the document]
Comment (irlevesque) [reply packages/backend/src/apps/todoist/triggers/get-tasks/get-tasks.ts]: OK, updated. I noticed that with `params` the `trim()` would throw an error if the parameter wasn't set. I'm not sure how safe it is to remove `trim()` but that was the only way I could find to cleanly allow the parameters to be optional.

**Comment by irlevesque** [comment]
Hi @barinali -- great feedback, thanks! Please see my comments above and let me know if there should be any more changes. I can take a look into the OAuth2 requirements as time allows, but perhaps that can be in a follow-up PR?

File: packages/backend/src/apps/todoist/actions/create-task/index.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (barinali) [reply packages/backend/src/apps/todoist/actions/create-task/index.ts]: It's already feasible. The global variable (`$`) in the dynamic data implementation can access its step parameters as can be seen at [GitHub - list a repository labels](https://github.com/automatisch/automatisch/blob/main/packages/backend/src/apps/github/dynamic-data/list-labels/index.ts#L16) sample. In the Todoist case, you should be able to access the project ID in the same approach. Automatisch will also show an error if the `projectId` field is empty when you define a `dependsOn` option in the `sectionId` field to imply `sectionId` field relies on the `projectId` field.

\- `dependsOn` definition sample; https://github.com/automatisch/automatisch/blob/main/packages/backend/src/apps/github/triggers/new-issues/index.ts#L65
\- A sample for defining a source to fetch dynamic data for the dropdown, https://github.com/automatisch/automatisch/blob/main/packages/backend/src/apps/github/triggers/new-issues/index.ts#L66-L79. In this sample, you'll see we send `parameters.repo` as an argument to make it available in the dynamic data implementation.

We can also bring this in a separate pull request later. I can also take care of this if you let me know if it's fine.

File: packages/backend/src/apps/todoist/triggers/get-tasks/get-tasks.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (barinali) [reply packages/backend/src/apps/todoist/triggers/get-tasks/get-tasks.ts]: We may use conditional chaining as in `($.step.parameters.filter as string)?.trim()`, which should trim defined and should not throw when it's `null`. What do you think?

**Comment by barinali** [comment]
> Hi @barinali -- great feedback, thanks! Please see my comments above and let me know if there should be any more changes. I can take a look into the OAuth2 requirements as time allows, but perhaps that can be in a follow-up PR?

Thanks a lot for the quick iteration of the remarks! I appreciate it. Once we conclude the new comments I wrote, I'd be happy to proceed with this branch.

Regarding the OAuth2, as you mentioned, we could do it in a separate pull request. I can also take care of that. Let me know how you'd prefer it.

File: packages/backend/src/apps/todoist/actions/create-task/index.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (irlevesque) [reply packages/backend/src/apps/todoist/actions/create-task/index.ts]: Thanks, that works perfectly.

**Comment by irlevesque** [comment]
Thanks again for the helpful feedback, which I've incorporated into this branch. 

> Regarding the OAuth2, as you mentioned, we could do it in a separate pull request. I can also take care of that. Let me know how you'd prefer it.

Sure, if you'd like to handle that, that would be great. My needs are met with a simple API token.

**Review by barinali — APPROVED** [review]
@irlevesque, I appreciate your contribution to Automatisch! Thanks a bunch.

As discussed, I'll rework the authentication with OAuth 2 later. At that point, you might need to "reconnect" your connection for Todoist, depending on how the migration works.

**Pull request merged**